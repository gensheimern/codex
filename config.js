module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
