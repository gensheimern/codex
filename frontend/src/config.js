const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;