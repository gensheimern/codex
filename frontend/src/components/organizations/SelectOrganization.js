import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import config from '../../config';
import JoinOrganization from '../organizations/JoinOrganization';
import AddOrganization from './AddOrganization';
import ListOrganizations from './ListOrganizations';

const NOT_SELECTED = -1;
const CREATE_NEW = 0;

export default class SelectOrganization extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			organizations: [],
			loaded: false,
			error: null,
			password: '',
			selected: NOT_SELECTED,
		};
		
		this.back = this.back.bind(this);
	}

	componentDidMount() {
		this.loadOrganizations();
	}

	loadOrganizations() {
		fetch(`${config.apiPath }/organization/`, {
            method: 'GET',
            headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
            },
		})
		.then((res) => {
            if(!res.ok) {
                throw new Error("Response not ok.");
            } else if(res.status !== 200) {
                throw new Error("An error occured.");
            }
            return res.json();
		})
		.then((res) => {
            this.setState({
				loaded: true,
				error: null,
				organizations: res,
			});
        }).catch((error) => {
			this.setState({
				loaded: true,
				error,
				organizations: [],
			});
        });
	}

	back() {
		this.setState({
			selected: NOT_SELECTED,
		});
	}

	render() {
		const organizations = this.state.organizations.filter((organization) => {
			return !this.props.myOrganizations.includes(organization.id);
		});

		const back = this.state.selected !== NOT_SELECTED
			? <FlatButton
				label="Back"
				primary={true}
				onClick={this.back}
			/>
			: null;

		const actions = [
			back,
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.props.close}
			/>,
		];

		let action = null;

		if (this.state.selected === NOT_SELECTED) {
			action = (
				<React.Fragment>
					<ListOrganizations
						organizations={organizations}
						onJoin={(selected) => {
							this.setState({
								selected,
							});
						}}
					/>	
					<RaisedButton
						label="Create new"
						primary={true}
						onClick={() => {
							this.setState({
								selected: CREATE_NEW,
							});
						}}
						style={{
							margin: '2%',
						}}
					/>
				</React.Fragment>	
			);
		}

		if (this.state.selected > CREATE_NEW) {
			action = (
				<JoinOrganization
					show={this.state.selected !== NOT_SELECTED}
					id={this.state.selected}
					organization={this.state.organizations.filter((element) => element.id === this.state.selected)[0]}
					saveChange={this.props.saveChange}
					close={this.props.close}
				/>
			);
		}

		if (organizations.length === 0) {
			action = (
				<p style={{
					width: '100%',
					textAlign: 'center',
				}}>No organizations yet. Create one.</p>
			);
		}

		if (this.state.selected === CREATE_NEW) {
			action = (
				<AddOrganization
					saveChange={this.props.saveChange}
					close={this.props.close}
				/>
			);
		}

		return (
			<Dialog
				title="Select Organization"
				modal={false}
				open={this.props.show}
				onRequestClose={this.props.close}
				autoScrollBodyContent={true}
				actions={actions}
			>
				{action}
			</Dialog>
		);
	}
}
