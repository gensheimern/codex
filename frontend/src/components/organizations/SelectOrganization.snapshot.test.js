/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import SelectOrganization from './SelectOrganization';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get( config.apiPath +'/organization/',
    {
        id: 8,
		name: "Das Team",
		description: "Der Test",
		admin: {
            id:7,
            firstName: "Max",
            name: "Mustermann",
            email: "max.mustermann@gmail.com",
            image: "",
            me: true,
		},
    });
});

describe('SelectOrganization Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <SelectOrganization organizations= {[]}/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});