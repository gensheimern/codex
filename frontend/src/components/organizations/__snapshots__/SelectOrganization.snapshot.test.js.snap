// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOrganization Snapshot renders 1`] = `null`;
